src/components/light/Light.jsx

```diff
@@ -0,0 +1,32 @@
+import React, { useReducer } from "react";
+import "./style.css";
+import lightOff from "./images/light-off.svg";
+import lightOn from "./images/light-on.svg";
+
+const Light = ({ name, state, id, dispatch }) => {
+  const getStateAsBoolean = (state) => {
+    return state === "on" ? true : false;
+  };
+
+  return (
+    <div className="light">
+      <div
+        className="light__item"
+        onClick={() => dispatch({ type: "toggle", payload: { id: id } })}
```

Comment:
asi bych si do komponenty Light neposílal funkci dispatch. To že je stav implementovaný pomocí useReducer v Lights je detail který Light nemusí vědět. Tj vem si příklad kdy se rozhodneš stav v Lights překopat z useReducer na useState. Tak musíš změnit 2 soubory, Lights a Light. Tj ideálně chceme aby Light nevědělo zbytečně moc informací o tom, jak je implementovaná Lights (jinak to znamená že jsou těsně svázané, "tighly coupled" a to většinou nechceme.) (koukni na Single Responsibility principle tady https://www.geeksforgeeks.org/solid-principle-in-programming-understand-with-real-life-examples/)

Tj udělal bych to tak, že Light by jako props žrala místo dispatch 2 funkce, onToggle a onRemove. Tady by pak bylo
```
onClick={onToggle}
```
a v Lights by bylo
```
        <Light
          ...
          onToggle={() => dispatch({ type: "toggle", payload: { id: id } })}
        />
```

Reply:
tím pádem Light neřeší jak přesně je implementovaný stav v Lights. Jeho jedinná povinnost je správně zavolat ten onToggle nebo onRemove callback